R/pkg/R/sparkR.R

```diff
@@ -558,16 +558,18 @@ sparkCheckInstall <- function(sparkHome, master) {
       message(msg)
       NULL
     } else {
-      if (!nzchar(master) || isMasterLocal(master)) {
+      if (isMasterLocal(master)) {
         msg <- paste0("Spark not found in SPARK_HOME: ",
                       sparkHome)
         message(msg)
         packageLocalDir <- install.spark()
         packageLocalDir
-      } else {
+      } else if (master == "yarn-client") {
```

Comment:
should this be changed to hard coded to "yarn-client"?
if I recall, master can be "yarn" and deploy mode can "client" or "cluster" whereas it is possible to have "yarn-client"/"yarn-cluster" - both ways should work.